src/main/java/net/minestom/server/entity/Player.java

```diff
@@ -271,6 +275,18 @@ public Player(@NotNull UUID uuid, @NotNull String username, @NotNull PlayerConne
         metadata.setNotifyAboutChanges(false);
     }
 
+    public void transfer(@NotNull InetSocketAddress address) {
+        this.transfer(address.getAddress().getHostAddress(), address.getPort());
+    }
+
+    public void transfer(@NotNull String host, int port) {
+        TransferPacket packet = new TransferPacket(host, port);
+        EventDispatcher.call(new PrePlayerTransferEvent(this, host, port));
```

Comment:
I think this can just be PlayerTransfer event, having pre and post seems excessive. It probably should be cancellable also